File: algo/src/main/java/com/hartwig/actin/algo/evaluation/reproduction/IsBreastfeeding.java
package com.hartwig.actin.algo.evaluation.reproduction;

import com.hartwig.actin.PatientRecord;
import com.hartwig.actin.algo.datamodel.Evaluation;
import com.hartwig.actin.algo.datamodel.EvaluationResult;
import com.hartwig.actin.algo.evaluation.EvaluationFactory;
import com.hartwig.actin.algo.evaluation.EvaluationFunction;
import com.hartwig.actin.clinical.datamodel.Gender;

import org.jetbrains.annotations.NotNull;

public class IsBreastfeeding implements EvaluationFunction {

    IsBreastfeeding() {
    }

    @NotNull
    @Override
    public Evaluation evaluate(@NotNull PatientRecord record) {
        if (record.clinical().patient().gender() == Gender.MALE) {
            return EvaluationFactory.unrecoverable()
                    .result(EvaluationResult.FAIL)
                    .addFailSpecificMessages("Patient is male thus won't be breastfeeding")
                    .addFailGeneralMessages("No breastfeeding")
                    .build();
        } else {
            return EvaluationFactory.unrecoverable()
                    .result(EvaluationResult.NOT_EVALUATED)
                    .addPassSpecificMessages("It is assumed that patient won't be breastfeeding")
                    .addPassGeneralMessages("Assumed no breastfeeding")
                    .build();
        }
    }
}

File: algo/src/main/kotlin/com/hartwig/actin/algo/evaluation/reproduction/IsBreastfeeding.kt
package com.hartwig.actin.algo.evaluation.reproduction

import com.hartwig.actin.algo.evaluation.EvaluationFactory
import com.hartwig.actin.algo.evaluation.EvaluationFunction
import com.hartwig.actin.datamodel.PatientRecord
import com.hartwig.actin.datamodel.algo.Evaluation
import com.hartwig.actin.datamodel.clinical.Gender

class IsBreastfeeding: EvaluationFunction {

    override fun evaluate(record: PatientRecord): Evaluation {
        return if (record.patient.gender == Gender.MALE) {
            EvaluationFactory.fail("No breastfeeding (is male)")
        } else {
            EvaluationFactory.notEvaluated("Currently assumed that patient is not breastfeeding")
        }
    }
}